packages/components/src/components/ui-toggle/ui-toggle.tsx

```diff
@@ -0,0 +1,287 @@
+import { Component, Prop, State, h, Watch, Event, EventEmitter } from '@stencil/core';
+
+/**
+ * Toogle switch component with various fetueres, multiple visual styles and TD integration.
+ * Link a direct property URL for plug-and-play device control.
+ * 
+ * @example Basic Usage
+ * ```html
+ * <ui-toggle variant="circle" state="active" label="Light"></ui-toggle>
+ * ```
+ * 
+ * @example TD Integration
+ * ```html
+ * <ui-toggle td-url="http://plugfest.thingweb.io/http-data-schema-thing/properties/bool" label="Test Device"></ui-toggle>
+ * ```
+ */
+@Component({
+  tag: 'ui-toggle',
+  styleUrl: 'ui-toggle.css',
+  shadow: true,
+})
+export class UiToggle {
+  /**
+   * Visual style variant of the toggle.
+   * - circle: Common pill-shaped toggle (default)
+   * - square: Rectangular toggle with square thumb
+   * - apple: iOS-style switch (bigger size, rounded edges)
+   * - cross: Shows × when off, ✓ when on with red background when off and green when on
+   * - neon: Glowing effect when active
+   */
+  @Prop() variant: 'circle' | 'square' | 'apple' | 'cross' | 'neon' = 'circle';
+
+  /**
+   * Current state of the toggle.
+   * - active: Toggle is on/active
+   * - disabled: Toggle cannot be clicked or interacted with
+   * - default: Toggle is off/inactive (default)
+   */
+  @Prop({ mutable: true }) state: 'active' | 'disabled' | 'default' = 'default';
+
+  /**
+   * Theme for the component.
+   */
+  @Prop() theme: 'light' | 'dark' = 'light';
+
+  /**
+   * Color scheme to match thingsweb webpage 
+   */
+  @Prop() color: 'primary' | 'secondary' | 'neutral' = 'primary';
+
+  /**
+   * Optional text label, to display text left to the toggle.
+   * When given, clicking the label will also toggle the switch.
+   */
+  @Prop() label?: string;
+
+  /**
+   * Direct URL of TD boolean properties to auto connect and interact with the device.
+   * @example
+   * ```
+   * td-url="http://plugfest.thingweb.io:80/http-data-schema-thing/properties/bool"
+   * ```
+   */
+  @Prop() tdUrl?: string;
+
+  /** Internal state tracking if toggle is on/off */
+  @State() isActive: boolean = true;
```

Comment:
if this is just an internal state should be private.

```suggestion
  /** Internal state tracking if toggle is on/off */
  @State() private isActive: boolean = true;
```

Reply:
Ok I will make internal state private afterwards